src/meta/src/rpc/ddl_controller.rs

```diff
@@ -1433,8 +1454,28 @@ impl DdlController {
                             dropping_sink_id: None,
                             updated_sink_catalogs,
                         },
+                        release_ctx.as_ref(),
                     )
                     .await?;
+                if let Some(release_ctx) = &release_ctx {
+                    // drop table associated source, we have already drop the objects in catalog above and need to run command to clean up the hummock entry.
+                    self.source_manager
+                        .apply_source_change(SourceChange::DropSource {
+                            dropped_source_ids: release_ctx.removed_source_ids.clone(),
+                        })
+                        .await;
+                    self.stream_manager
+                        .drop_streaming_jobs(
```

Comment:
Why did it trigger another barrier to drop? All old actors should have been dropped in previous replace barrier processing. I guess the purpose is to unregister the source state table in the hummock? If so, we need to add this logic after the replace barrier is collected.